contrib/pkg/createcluster/create.go

```diff
@@ -325,20 +331,72 @@ func (o *Options) getResourceHelper() (*resource.Helper, error) {
 func (o *Options) GenerateObjects() ([]runtime.Object, error) {
 	result := []runtime.Object{}
 
-	pullSecret, err := o.generatePullSecret()
+	pullSecret, err := o.getPullSecret()
 	if err != nil {
 		return nil, err
 	}
 
-	cd, err := o.GenerateClusterDeployment(pullSecret)
+	pullSecretSecret, err := o.generatePullSecret(pullSecret)
 	if err != nil {
 		return nil, err
 	}
 
-	if err := o.cloudProvider.addPlatformDetails(o, cd); err != nil {
+	cd, err := o.GenerateClusterDeployment(pullSecretSecret)
+	if err != nil {
+		return nil, err
+	}
+
+	sshPublicKey, err := o.getSSHPublicKey()
+	if err != nil {
+		return nil, err
+	}
+	controlPlanePool := &installertypes.MachinePool{
```

Comment:
Don't need `controlPlanePool` and `computePool` variables. This info can be set directly in `installConfig` when it is initialized.